src/txn/txn.c

```diff
@@ -1251,6 +1273,10 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
     WT_ASSERT(session, F_ISSET(txn, WT_TXN_RUNNING));
     WT_ASSERT(session, !F_ISSET(txn, WT_TXN_ERROR) || txn->mod_count == 0);
 
+    /* Configure the timeout for this commit operation. */
+    if (cfg != NULL)
```

Comment:
Would it be simpler to move this NULL check into `txn_config_operation_timeout()`? In general, I prefer to let callees validate parameters, as doing it in one place is less error-prone. 

The question, I guess, is the call in `wt_txn_config()`, which doesn't have a check. And in looking at that, I see it's not needed because the one caller of `wt_txn_config()` checks that the configuration string is non-NULL before calling.  So I obviously can't argue from precedent that the callee should always do the parameter checking! :-) 

Just offering this as food for thought.  No requirement to change this.

Reply:
I had the same thought when I looked at this diff this morning (but hadn't done a review yet). I would, in fact, suggest that the caller of `wt_txn_config` and `wt_txn_config` be changed in the same way.